File: YYReader/Parsing/ParsedChapterPage.swift
import Foundation

struct ParsedChapterPage: Sendable {
    let title: String
    let bookTitle: String?
    let author: String?
    let paragraphs: [String]
    let catalogURL: URL?
    let previousChapterURL: URL?
    let nextChapterURL: URL?
    let nextPageURL: URL?
}

File: windows/YYReader.Windows.Core/Parsing/ParsedChapterPage.cs
namespace YYReader.Windows.Core.Parsing;

public sealed record ParsedChapterPage(
    string Title,
    string? BookTitle,
    string? Author,
    IReadOnlyList<string> Paragraphs,
    Uri? CatalogUrl,
    Uri? PreviousChapterUrl,
    Uri? NextChapterUrl,
    Uri? NextPageUrl);
